### DISCUSSION TITLE
电脑的哪些配件更换会导致识别码更换？

### DISCUSSION THREAD

**Discussion by SALTWOOD** [Meloong-Git/PCL#4070] [提问] [answered]
最近准备升级电脑，但是因为学生党财力有限，只能分阶段更换
这就导致一个问题：每个阶段之间一大段时间，不是一次性换掉所有硬件，这肯定会导致识别码更换，这样账号要重新登录，主题要重新解锁，活跃橙要再找龙猫申请，于是就想先吃一颗定心丸。

在此问一下各位 Collaborator 和龙猫，哪些硬件更换会导致识别码更换哪些不会？这样我就可以把会影响识别码的硬件先换上，然后就可以先把这一对麻烦的事搞定

1. 主板和内存换掉
2. 显卡换掉
3. 硬盘换掉，重装系统，但是备份注册表

劳烦问一下哪些会导致识别码更换。谢谢～！

**Comment by HomoobyDX** [comment]
个人认为，你把注册表备份好了，你对PCL2的设置就都会留下（并且PCL2里本来就写了赞助者可通过注册表备份来保存设置）

**Comment by WForst-Breeze** [comment]
只要重装了系统，那就会进行识别码变更，隐藏主题、账号密码就是手动需要重新恢复的，设置可以进行备份迁移。其他操作一般而言不会变更识别码。（参阅：#1850）

---

#### · 把主板和内存换掉

> 参考：[在更换硬件后重新激活 Windows](https://support.microsoft.com/zh-cn/windows/%E5%9C%A8%E6%9B%B4%E6%8D%A2%E7%A1%AC%E4%BB%B6%E5%90%8E%E9%87%8D%E6%96%B0%E6%BF%80%E6%B4%BB-windows-2c0e962a-f04c-145b-6ead-fb3fc72b6665#ID0EBD=Windows_10)
>
> 在安装 Windows 10 时，数字许可证本身与设备的硬件相关联。 如果你对设备进行重大的硬件更改（例如更换母板），Windows 将无法找到与设备匹配的许可证，因此你需要重新激活 Windows 才能使其正常运行。

更换主板后 **可能需要** 重装系统，如此识别码即会变更。

#### · 显卡换掉

一般不会变更识别码。

#### · 硬盘换掉，重装系统，但是备份注册表

识别码会变更，需要重新操作恢复相关内容。

可能有误，欢迎指正 orz

**Comment by Silverteal** [comment, marked as answer]
两种情况中的至少一种发生会使识别码改变，且都会导致部分配置丢失。
1.注册表的`Identify`键丢失或改变
2.设备的SMBIOS UUID改变

**Reply by SALTWOOD** [reply]
那么是不是意味着我更换完主板之后会且只会变动一次？

重装系统，我肯定会把 `Identify` 备份的

**Reply by Silverteal** [reply]
按我的理解，应该是的。~~但是实际上也出现过插了个耳机之后识别码就变了的情况（issue #3316）~~ 实际上是某个旧版本修改了计算方式导致的。
能确定的是，主板更换一定会导致识别码变化，至于其他硬件的更换，可能取决于主板如何计算SMBIOS UUID，但多半是不会变的。

**Reply by wuliaodexiaoluo** [reply]
> 按我的理解，应该是的。但是实际上也出现过插了个耳机之后识别码就变了的情况（issue #忘了）。 

这锅丢给龙猫，2.6.13 刷的识别码
Issue #3316

**Comment by LTCatt** [comment]
需要活跃橙的话再申请一下就行，无所谓啦
